In [1]:
# Step 1 Import library

import pandas as pd

In [2]:
# Step 2 Read datasets

df= pd.read_csv('https://github.com/YBIFoundation/Dataset/raw/main/Bank%20Churn%20Modelling.csv')


In [3]:
df.head()

CustomerId   Surname  CreditScore Geography  Gender  Age  Tenure  \
0    15634602  Hargrave          619    France  Female   42       2   
1    15647311      Hill          608     Spain  Female   41       1   
2    15619304      Onio          502    France  Female   42       8   
3    15701354      Boni          699    France  Female   39       1   
4    15737888  Mitchell          850     Spain  Female   43       2   

     Balance  Num Of Products  Has Credit Card  Is Active Member  \
0       0.00                1                1                 1   
1   83807.86                1                0                 1   
2  159660.80                3                1                 0   
3       0.00                2                0                 0   
4  125510.82                1                1                 1   

   Estimated Salary  Churn  
0         101348.88      1  
1         112542.58      0  
2         113931.57      1  
3          93826.63      0  
4          79084.10      0

In [4]:
# One hot encoding

df1= pd.get_dummies(df, columns=['Geography','Gender'])

In [5]:
df1.head()

CustomerId   Surname  CreditScore  Age  Tenure    Balance  Num Of Products  \
0    15634602  Hargrave          619   42       2       0.00                1   
1    15647311      Hill          608   41       1   83807.86                1   
2    15619304      Onio          502   42       8  159660.80                3   
3    15701354      Boni          699   39       1       0.00                2   
4    15737888  Mitchell          850   43       2  125510.82                1   

   Has Credit Card  Is Active Member  Estimated Salary  Churn  \
0                1                 1         101348.88      1   
1                0                 1         112542.58      0   
2                1                 0         113931.57      1   
3                0                 0          93826.63      0   
4                1                 1          79084.10      0   

   Geography_France  Geography_Germany  Geography_Spain  Gender_Female  \
0                 1                  0                0              1   
1                 0                  0                1              1   
2                 1                  0                0              1   
3                 1                  0                0              1   
4                 0                  0                1              1   

   Gender_Male  
0            0  
1            0  
2            0  
3            0  
4            0

In [6]:
df1.columns

Index(['CustomerId', 'Surname', 'CreditScore', 'Age', 'Tenure', 'Balance',
       'Num Of Products', 'Has Credit Card', 'Is Active Member',
       'Estimated Salary', 'Churn', 'Geography_France', 'Geography_Germany',
       'Geography_Spain', 'Gender_Female', 'Gender_Male'],
      dtype='object')

In [7]:
# Step 3 Define x & y

y= df1['Churn']

X= df1[[ 'CreditScore', 'Age', 'Tenure', 'Balance','Num Of Products', 'Has Credit Card', 'Is Active Member',
       'Estimated Salary', 'Geography_France', 'Geography_Germany','Geography_Spain', 'Gender_Female', 'Gender_Male']]

In [8]:
# Step 4 Train test split

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test= train_test_split(X,y, random_state=2529)

In [9]:
# Step 5 Select a model

from sklearn.ensemble import RandomForestClassifier
model= RandomForestClassifier()

In [10]:
# step 6 Training the model

model.fit(X_train, y_train)

RandomForestClassifier()

In [11]:
# Step 7 Predict the model

y_pred= model.predict(X_test)

In [12]:
## step 8 Evaluation

from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.87      0.97      0.92      2012
           1       0.77      0.43      0.55       488

    accuracy                           0.86      2500
   macro avg       0.82      0.70      0.73      2500
weighted avg       0.85      0.86      0.85      2500



**Model is considered to be bad since prediction is 55 % only**